### ISSUE TITLE
autodoc raises NotImplementedError for non-solidity domain

### ISSUE THREAD

**Issue by oberstet** [cag/sphinxcontrib-soliditydomain#4]
Issue: cannot document Solidity and Python sources in the same project.

---

With `sphinxcontrib.soliditydomain` added to the extensions, auto-documenting Solidity sources works, but auto-documenting a Python class fails with:

```
reading sources... [ 80%] api/xbrpy                                                                                                                                                                                                                                                
Exception occurred:
  File "/home/oberstet/cpy3_1/lib/python3.7/site-packages/sphinx/ext/autodoc/__init__.py", line 296, in can_document_member
    raise NotImplementedError('must be implemented in subclasses')
NotImplementedError: must be implemented in subclasses
The full traceback has been saved in /tmp/sphinx-err-30qmik_i.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
Makefile:156: recipe for target 'docs' failed
make: *** [docs] Error 2
```

I used the following trivial class to track down this issue:

```python
class Test(object):
    def __init__(self, foo):
        """

        :param foo:
        """
        self.foo = foo

    def add2(self, x, y):
        """

        :param x:
        :param y:
        :return:
        """
        return x + y
```

and the following directive in a RST file:

```rst
.. autoclass:: xbr.Test
    :members: add2
```

---

Could it be that `sphinxcontrib.soliditydomain` is overwriting something that it shouldn't?

**Comment by cag** [comment]
Thanks for the report! I'll have a look soon, but that would be my first guess as well. :thinking: 

@oberstet As part of an effort to address this, I've started adding a contributing guide which refers to the python implementation of this plugin in the documentation. A first cut:

```rst
.. autoclass:: sphinxcontrib.soliditydomain.SolidityDomain
```

works just fine. However,

```rst
.. autoclass:: sphinxcontrib.soliditydomain.SolidityDomain
    :members: resolve_xref
```

reproduces the error which you have experienced, as does

```rst
.. autoclass:: sphinxcontrib.soliditydomain.SolidityDomain
    :members:
```

Also, commenting out the plugin's inclusion into the documentation build allows the python autodoc to succeed. I had thought that I had taken care in not stepping on toes in my code, but apparently I'm wrong!

My excuse is that I could not find any material on how to write a proper Sphinx plugin, so I am spelunking and improvising as I go :sweat_smile: ... Anyway, this is a real issue... I'll report back soon hopefully.

**Comment by cag** [comment]
Should be fixed in 0.4.1

**Comment by oberstet** [comment]
I can confirm the fix works (using 0.4.2): documentation is auto-generated for both Solidity and Python domain objects. Awesome, thanks a lot for the very quick response! 

> Sidenote: I stumbled across your (I presume) blog post https://blog.gnosis.pm/solidity-delegateproxy-contracts-e09957d0f201. Comprehensive, impressive level of detail. Also thanks a lot for that! Proxy stuff is on our backlog (contract upgradability and saving gas)
